spectral_cube/base_class.py

```diff
@@ -1,16 +1,23 @@
 from astropy import units as u
 from astropy import log
+import numpy as np
 
 from . import wcs_utils
+from . import cube_utils
+from .utils import cached, WCSCelestialError
+
+__doctest_skip__ = ['SpatialCoordMixinClass.world']
 
 DOPPLER_CONVENTIONS = {}
 DOPPLER_CONVENTIONS['radio'] = u.doppler_radio
 DOPPLER_CONVENTIONS['optical'] = u.doppler_optical
 DOPPLER_CONVENTIONS['relativistic'] = u.doppler_relativistic
 
 
-
 class BaseNDClass(object):
+
+    _cache = {}
```

Comment:
Needed for caching. `BaseSpectralCube` does this as well: https://github.com/radio-astro-tools/spectral-cube/blob/master/spectral_cube/spectral_cube.py#L133.

Switching to `lazyproperty` seems like the better route. Maybe that should be separate PR?